File: remove_theme.py
import re
import os

files = [
    'Resources.html', 'shops.html', 'tier-list.html', 'Locations.html',
    'quests.html', 'manufacture.html', 'roadmap.html', 'index.html',
    'monsters.html', 'Item.html'
]

for filename in files:
    filepath = os.path.join(r'C:\Users\cloud\Downloads\Steam\WLO\Ress\Project', filename)
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original = content
    
    # 1. Remove CSS: /* Theme Toggle */ + .theme-toggle, .theme-toggle-ball, .theme-toggle.light CSS
    content = re.sub(
        r'\s*/\*\s*Theme\s+Toggle\s*\*/\s*'
        r'\.theme-toggle\s*\{[^}]*\}\s*'
        r'\.theme-toggle-ball\s*\{[^}]*\}\s*'
        r'\.theme-toggle\.light\s+\.theme-toggle-ball\s*\{[^}]*\}',
        '', content, flags=re.DOTALL
    )
    
    # 2. Remove /* Light mode styles - pastel theme */ comment
    content = re.sub(
        r'\n\s*/\*\s*Light mode styles\s*[-–—]?\s*pastel theme\s*\*/',
        '', content
    )
    
    # 3. Remove all body.light-mode CSS rules (multi-property)
    content = re.sub(
        r'\n\s*body\.light-mode\s+[^{]*?\{[^}]*\}',
        '', content, flags=re.DOTALL
    )
    
    # 4. Remove any remaining .light-mode CSS rules
    content = re.sub(
        r'\n\s*\.[a-zA-Z0-9_-]*light-mode[a-zA-Z0-9_-]*\s*\{[^}]*\}',
        '', content, flags=re.DOTALL
    )
    
    # 5. Remove HTML: theme toggle element
    content = re.sub(
        r'\s*<!--\s*Theme\s+Toggle\s*-->\s*'
        r'<div\s+class="theme-toggle"[^>]*>\s*'
        r'<div\s+class="theme-toggle-ball">\s*'
        r'<i\s+class="fas\s+fa-moon[^"]*"[^>]*>\s*</i>\s*'
        r'</div>\s*'
        r'</div>',
        '', content, flags=re.DOTALL
    )
    
    # 6. Remove JavaScript: theme toggle script blocks
    content = re.sub(
        r'<script>\s*'
        r'//\s*Theme\s+Toggle\s*'
        r'\s*const\s+themeToggle\s*=\s*document\.getElementById\([\'"]themeToggle[\'"]\)\s*;'
        r'\s*const\s+body\s*=\s*document\.body\s*;'
        r'.*?'
        r'\}\s*\)\s*;'
        r'\s*</script>',
        '', content, flags=re.DOTALL
    )
    
    # 7. Clean up multiple blank lines
    content = re.sub(r'\n{4,}', '\n\n', content)
    
    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Modified: {filename}")
    else:
        print(f"No changes: {filename}")

File: remove_theme.pl
#!/usr/bin/perl
use strict;
use warnings;

my @files = qw(
    Resources.html shops.html tier-list.html Locations.html quests.html
    manufacture.html roadmap.html index.html monsters.html Item.html
);

foreach my $file (@files) {
    open(my $fh, '<:encoding(UTF-8)', $file) or die "Cannot open $file: $!";
    my $content = do { local $/; <$fh> };
    close($fh);
    
    my $original = $content;
    
    # 1. Remove CSS: /* Theme Toggle */ comment + .theme-toggle, .theme-toggle-ball, .theme-toggle.light CSS
    $content =~ s{
        \s*/[*]\s*Theme\s+Toggle\s*[*]/\s*
        [.](?:theme-toggle|theme-toggle-ball)\s*[{][^}]*[}]\s*
        [.](?:theme-toggle)[.]light\s+[.](?:theme-toggle-ball)\s*[{][^}]*[}]
    }{}gsx;
    
    # 2. Remove all body.light-mode CSS rules (including multi-property blocks)
    $content =~ s{\n\s*body[.]light-mode\s+(?:[^{]*?)\{(?:[^}]*)\}}{}gs;
    
    # 3. Remove /* Light mode styles - pastel theme */ comment
    $content =~ s{\n\s*/[*]\s*Light mode styles\s*[-–—]?\s*pastel theme\s*[*]/}{}g;
    
    # 4. Remove any remaining .light-mode CSS rules
    $content =~ s{\n\s*[.](?:[a-zA-Z0-9_-]*light-mode[a-zA-Z0-9_-]*)\s*[{][^}]*[}]}{}gs;
    
    # 5. Remove HTML: theme toggle element
    $content =~ s{
        \s*<!--\s*Theme\s+Toggle\s*-->\s*
        <div\s+class="theme-toggle"[^>]*>\s*
        <div\s+class="theme-toggle-ball">\s*
        <i\s+class="fas\s+fa-moon[^"]*"[^>]*>\s*</i>\s*
        </div>\s*
        </div>
    }{}gsx;
    
    # 6. Remove JavaScript: theme toggle script blocks
    $content =~ s{<script>\s*
        //\s*Theme\s+Toggle\s*
        \s*const\s+themeToggle\s*=\s*document[.]getElementById[(]themeToggle[)]\s*;
        \s*const\s+body\s*=\s*document[.]body\s*;
        .*?
        \}\s*\)\s*;
        \s*</script>
    }{}gsx;
    
    # 7. Clean up multiple blank lines (more than 3 newlines)
    $content =~ s{\n{4,}}{\n\n}g;
    
    if ($content ne $original) {
        open(my $wfh, '>:encoding(UTF-8)', $file) or die "Cannot write $file: $!";
        print $wfh $content;
        close($wfh);
        print "Modified: $file\n";
    } else {
        print "No changes: $file\n";
    }
}
